src/core/lib/iomgr/endpoint_cfstream.cc

```diff
@@ -365,15 +365,17 @@ grpc_endpoint* grpc_cfstream_endpoint_create(
   CFSTREAM_HANDLE_REF(ep_impl->stream_sync, "endpoint create");
 
   ep_impl->peer_string = peer_string;
+  grpc_resolved_address resolved_local_addr;
+  resolved_local_addr.len = sizeof(resolved_local_addr.addr);
   const int* native_handle =
       reinterpret_cast<const int*>(CFReadStreamCopyProperty(
           ep_impl->read_stream, kCFStreamPropertySocketNativeHandle));
-  grpc_resolved_address resolved_local_addr;
-  resolved_local_addr.len = sizeof(resolved_local_addr.addr);
-  if (getsockname(*native_handle,
-                  reinterpret_cast<sockaddr*>(resolved_local_addr.addr),
-                  &resolved_local_addr.len) < 0) {
+  if (native_handle &&
```

Comment:
This changes the control flow such that we don't call `getsockname()` at all if `native_handle` is null.  Is that what we want?

I would think that a better way to fix this would just be to say `if (native_handle != nullptr) CFRelease(native_handle)` inside of the `if (getsockname(...)) {` block.

Reply:
getsockname will set resolved_local_addr based on the current address to which the native socket handle is bound. So if the native_handle is null, I think the call wouldn't succeed anyway. Plus we would be dereferencing a nullptr.